pkg/cmd/openshift-apiserver/openshiftapiserver/config.go

```diff
@@ -0,0 +1,294 @@
+	//SecureServing *SecureServingInfo
+	//Authentication AuthenticationInfo
+	//Authorization AuthorizationInfo
+	//LoopbackClientConfig *restclient.Config
+	// this is set after the options are overlayed to get the authorizer we need.
+	//AdmissionControl      admission.Interface
+	//ReadWritePort int
+	//PublicAddress net.IP
+
+	// these are defaulted sanely during complete
+	//DiscoveryAddresses discovery.Addresses
+
+	genericConfig.CorsAllowedOriginList = openshiftAPIServerConfig.CORSAllowedOrigins
+	genericConfig.Version = &openshiftVersion
+	// we don't use legacy audit anymore
+	genericConfig.LegacyAuditWriter = nil
+	genericConfig.AuditBackend = backend
+	genericConfig.AuditPolicyChecker = policyChecker
+	genericConfig.ExternalAddress = "apiserver.openshift-apiserver.svc"
+	genericConfig.BuildHandlerChainFunc = OpenshiftHandlerChain
+	genericConfig.LegacyAPIGroupPrefixes = configprocessing.LegacyAPIGroupPrefixes
+	genericConfig.RequestInfoResolver = configprocessing.OpenshiftRequestInfoResolver()
+	genericConfig.OpenAPIConfig = configprocessing.DefaultOpenAPIConfig(*openshiftAPIServerConfig)
+	genericConfig.SwaggerConfig = defaultSwaggerConfig()
+	genericConfig.RESTOptionsGetter = restOptsGetter
+	// previously overwritten.  I don't know why
+	genericConfig.RequestTimeout = time.Duration(60) * time.Second
+	genericConfig.MinRequestTimeout = openshiftAPIServerConfig.ServingInfo.RequestTimeoutSeconds
+	genericConfig.MaxRequestsInFlight = openshiftAPIServerConfig.ServingInfo.MaxRequestsInFlight
+	genericConfig.MaxMutatingRequestsInFlight = openshiftAPIServerConfig.ServingInfo.MaxRequestsInFlight / 2
+	genericConfig.LongRunningFunc = configprocessing.IsLongRunningRequest
+
+	// I'm just hoping this works.  I don't think we use it.
+	//MergedResourceConfig *serverstore.ResourceConfig
+
+	servingOptions, err := configprocessing.ToServingOptions(openshiftAPIServerConfig.ServingInfo)
+	if err != nil {
+		return nil, err
+	}
+	if err := servingOptions.ApplyTo(&genericConfig.Config); err != nil {
+		return nil, err
+	}
+	authenticationOptions := genericapiserveroptions.NewDelegatingAuthenticationOptions()
+	authenticationOptions.RemoteKubeConfigFile = openshiftAPIServerConfig.MasterClients.OpenShiftLoopbackKubeConfig
+	if err := authenticationOptions.ApplyTo(&genericConfig.Authentication, genericConfig.SecureServing, genericConfig.OpenAPIConfig); err != nil {
+		return nil, err
+	}
+	authorizationOptions := genericapiserveroptions.NewDelegatingAuthorizationOptions()
+	authorizationOptions.RemoteKubeConfigFile = openshiftAPIServerConfig.MasterClients.OpenShiftLoopbackKubeConfig
+	if err := authorizationOptions.ApplyTo(&genericConfig.Authorization); err != nil {
+		return nil, err
+	}
+
+	informers, err := NewInformers(kubeInformers, kubeClientConfig, genericConfig.LoopbackClientConfig)
+	if err != nil {
+		return nil, err
+	}
+	if err := informers.GetOpenshiftUserInformers().User().V1().Groups().Informer().AddIndexers(cache.Indexers{
```

Comment:
@deads2k i wonder if this should be moved to the controller that use this index?

Reply:
> @deads2k i wonder if this should be moved to the controller that use this index?

it's in the server as I recall.  Perhaps, but not here.